web/src/settings_users.js

```diff
@@ -94,6 +113,52 @@ function failed_listing_users() {
     blueslip.error("Error while listing users for user_id", {user_id});
 }
 
+function add_value_to_filters(section, key, value) {
+    section.filters[key] = value;
+    // This hard_redraw will rerun the relevant predicate function
+    // and in turn apply the new filters.
+    section.list_widget.hard_redraw();
+}
+
+function role_selected_handler(event, dropdown, widget) {
+    event.preventDefault();
+    event.stopPropagation();
+
+    const role_code = Number($(event.currentTarget).attr("data-unique-id"));
+    if (widget.widget_name === section.active.dropdown_widget_name) {
+        add_value_to_filters(section.active, "role_code", role_code);
+    } else if (widget.widget_name === section.deactivated.dropdown_widget_name) {
+        add_value_to_filters(section.deactivated, "role_code", role_code);
+    }
+
+    dropdown.hide();
+    widget.render();
+}
+
+function get_roles() {
+    return [
+        {unique_id: "0", name: $t({defaultMessage: "All roles"})},
+        ...Object.values(settings_config.user_role_values).map((user_role_value) => ({
+            unique_id: user_role_value.code.toString(),
+            name: user_role_value.description,
```

Comment:
I guess we might need to add a `description_plural` field in `settings_config.user_role_values` if we want the items in the dropdown to be plural, e.g., "Admins".